### PR TITLE
Adding auxclick as event handler

### PR DESCRIPTION
**Pull request by TorstenDittmann**
- Added auxclick as event
- Separate function for EventListener

Might also add EventListener for 'contextmenu' for "Open in new tab" with right click.

### CODE REVIEW

File: Router.svelte [hunk 1]
```diff
@@ -76,7 +76,7 @@ export function push(location) {
 
     // Execute this code when the current call stack is complete
     setTimeout(() => {
-        window.location.hash = '#' + location
+        window.location.hash = location
```
Comment (ItalyPaleAle) [inline Router.svelte]: This might be breaking. Should add an “if” statement to add a # if not present already.

Comment (TorstenDittmann) [reply Router.svelte]: Ah yeah, if function is used in code it can break. So i just concat a # if not present.

File: Router.svelte [hunk 2]
```diff
@@ -76,7 +76,7 @@ export function push(location) {
 
     // Execute this code when the current call stack is complete
     setTimeout(() => {
-        window.location.hash = location
+        window.location.hash = location.charAt(0) == '#' ? location : '#' + location
```
Comment (ItalyPaleAle) [inline Router.svelte]: Wait this won't work either. In line 73 we are checking to make sure that location.charAt(0) is '/'.

Need to update line 73 to:

````js
if (!location || location.length < 1 || (location.charAt(0) != '/' && location.indexOf('#/') !== 0)) {
````

**Comment by ItalyPaleAle** [comment]
Sorry for the delay in reviewing this. It's mostly ok. A couple more tweaks but I'll do that after merging. Thanks!

**Pull request merged**